### ISSUE TITLE
node build fails on OS X 10.11

### ISSUE THREAD

**Issue by mologie** [Homebrew/homebrew#40653]
```
$ brew install node --with-icu4c --with-openssl   
==> Downloading https://nodejs.org/dist/v0.12.4/node-v0.12.4.tar.gz
Already downloaded: /Library/Caches/Homebrew/node-0.12.4.tar.gz
==> ./configure --prefix=/Users/oliver/.local/Cellar/node/0.12.4 --without-npm --with-intl=system-icu --shared-openssl
==> make install
  ./gyp-mac-tool flock /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/linker.lock clang++ -Wl,-search_paths_first -mmacosx-version-min=10.5 -arch x86_64 -L/private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release  -o "/private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/mksnapshot" /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/obj.target/mksnapshot/deps/v8/src/mksnapshot.o /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/libv8_base.a /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/libv8_nosnapshot.a /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/libv8_libplatform.a /private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/libv8_libbase.a -L/Users/oliver/.local/Cellar/openssl/1.0.2a-1/lib -lssl -lcrypto -L/Users/oliver/.local/Cellar/icu4c/55.1/lib -licui18n -licuuc -licudata
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [/private/tmp/node20150613-90188-1v83nmc/node-v0.12.4/out/Release/mksnapshot] Error 1
make: *** [node] Error 2
```

This is likely an upstream issue in `node-gyp`: Apple removed the deprecated symlinks from `libgcc*.dylib` to `libSystem.B.dylib`.

Note that I have Homebrew installed under `$HOME/.local`, but considering the nature of the error, I would not deem this relevant. Installation with the same setup succeeds on OS X 10.10.

Harmless workaround for us early adopters:

```
$ cd /usr/local/lib
$ sudo ln -s ../../lib/libSystem.B.dylib libgcc_s.10.5.dylib
$ sudo ln -s ../../lib/libSystem.B.dylib libgcc_s.10.4.dylib
```

**Comment by mistydemeo** [comment]
Apple appears to be shipping `.tbd` files for `libgcc_s.10.{4,5}`.

At least on my 10.10 system (upgraded from previous OS Xs), `libgcc_s.10.4.dylib` is not a symlink to libSystem the way that `libgcc_s.1.dylib` is.

**Comment by mistydemeo** [comment]
Also, for reference, I was able to install node on 10.11 using the Xcode 7 CLT without an issue. Maybe there's an issue with your setup? Can you provide the output of `brew gist-logs node` please?

**Comment by mologie** [comment]
You are right, it looks like the reason this worked before is that the required symlinks exists in the OS X 10.10 SDK:

```
# oliver at Olivers-MacBook-Pro.fritz.box in /Applications/Xcode.app/Contents [19:18:15]
$ find . -iname 'libgcc_s.10.4*' 
./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s.10.4.dylib
./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/usr/lib/libgcc_s.10.4.dylib

# oliver at Olivers-MacBook-Pro.fritz.box in /Applications/Xcode.app/Contents [19:18:27]
$ ls -l ./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s*
lrwxr-xr-x  1 root  wheel     17 Jun 10 22:35 ./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s.1.dylib -> libSystem.B.dylib
lrwxr-xr-x  1 root  wheel     19 Jun 10 22:35 ./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s.10.4.dylib -> libgcc_s.10.5.dylib
-rwxr-xr-x  1 root  wheel  30976 Sep  9  2014 ./Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s.10.5.dylib
```

In the Xcode beta, they are gone:

```
# oliver at Olivers-MacBook-Pro.fritz.box in /Applications/Xcode-beta.app [19:20:03]
$ find . -iname 'libgcc_s.10.4*'                                                              
./Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk/usr/lib/libgcc_s.10.4.tbd
./Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/lib/libgcc_s.10.4.tbd
```

The Xcode stable version is my current command-line SDK according to `xcode-select -p`. I am unfamiliar with what `tbd` files actually are - this is the first time I stumbled upon them and Google is not very helpful on the issue.

**Comment by mistydemeo** [comment]
> I am unfamiliar with what tbd files actually are - this is the first time I stumbled upon them and Google is not very helpful on the issue.

They're new to Xcode 7, and I don't believe there is any public documentation on them yet.

**Comment by mologie** [comment]
That is probably the reason why the Xcode 6 toolchain did not pick these libraries/links up in the first place. This issue may just "go away" once Xcode 7 ships as default toolchain for OS X 10.11.

Using the Xcode 7 toolchain, everything worked as expected:

```
$ brew rm node
Error: No such keg: /Users/oliver/.local/Cellar/node

$ sudo xcode-select -s /Applications/Xcode-beta.app/Contents/Developer
Password:

$ xcode-select -p
/Applications/Xcode-beta.app/Contents/Developer

$ clang -v
Apple LLVM version 7.0.0 (clang-700.0.53)
Target: x86_64-apple-darwin15.0.0
Thread model: posix

$ brew install node
==> Downloading https://nodejs.org/dist/v0.12.4/node-v0.12.4.tar.gz
Already downloaded: /Library/Caches/Homebrew/node-0.12.4.tar.gz
==> ./configure --prefix=/Users/oliver/.local/Cellar/node/0.12.4 --without-npm --without-ssl2 --without-ssl3
==> make install
==> Downloading https://registry.npmjs.org/npm/-/npm-2.10.1.tgz
Already downloaded: /Library/Caches/Homebrew/node--npm-2.10.1.tgz
==> ./configure --prefix=/Users/oliver/.local/Cellar/node/0.12.4/libexec/npm
==> make install
==> Caveats
Bash completion has been installed to:
  /Users/oliver/.local/etc/bash_completion.d
==> Summary
/Users/oliver/.local/Cellar/node/0.12.4: 2590 files, 28M, built in 4.4 minutes
```

Thank you for the quick feedback - lesson learned, early access software tends to bite :). Shall we leave this bug report open for future reference, or close right away?

**Comment by mistydemeo** [comment]
Ah, yep, that would be the reason. We don't support Xcode 6 on 10.11; you need to use Xcode 7 to build software on 10.11. I'll close this then, thanks!

**Comment by mikemcquaid** [comment]
@mistydemeo Perhaps we should hard-error in that case?

**Comment by mistydemeo** [comment]
@mikemcquaid After #40536 is merged, maybe? We'd get a doctor warning there but hard erroring might be better than continuing to field these reports.

**Comment by DomT4** [comment]
@mistydemeo @mikemcquaid https://github.com/DomT4/homebrew/commit/34e4b640747040b40649d848a59ffcd7ea117acc

**Comment by mikemcquaid** [comment]
@DomT4 :heart:

**Comment by galxy25** [comment]
:+1:  for @mologie writeup. Saved me today.

**Comment by mrtorks** [comment]
Saved my ass and I do believe we should keep it open.

**Comment by mologie** [comment]
This issue should remain closed. As discussed in the comments, the issue was with my setup: Xcode 6 is not supported on OS X 10.11. Install Xcode 7 and configure Xcode 7 as default toolchain. With Xcode 7, node builds just fine without the workaround mentioned in my first post.

**Comment by paulhybryant** [comment]
I am seeing this error in my OSX 10.10 too.

**Comment by snikch** [comment]
This occurred to me too in OSX 10.10 after I downloaded an updated command line tools. Updating Xcode to 7 fixed the issue for me too.

**Comment by JoeyCarson** [comment]
Hate to ask on this thread, but I'm a little desperate.  I'm having the same build issue for libjpeg-turbo since updating to Xcode 7 and El Captian.  The comment above mentions to set Xcode 7 as the default toolchain.  What do you mean by setting it as the default toolchain?  For instance, I'm using a configure and make style build for libjpeg-turbo.  Is there another step I need to take to make this work with Xcode 7?

**Comment by DomT4** [comment]
@JoeyCarson Open a new issue with `brew gist-logs libjpeg-turbo` if you like.

**Comment by JoeyCarson** [comment]
Well this isn't really dependent on brew, I'm building libjpeg-turbo from svn trunk for iOS and simulator.  But when building it for the simulator, the same issue occurs, configure fails because it can't find libgcc.  This thread mentions that the issue is fixed by setting Xcode 7 as the default toolchain.  I don't quite understand that jargon there.  I have the Xcode iPhoneSimulator.sdk set as the sysroot, perhaps that means the same thing.  I can't tell because no one wrote a specific example of what the solution was.

**Comment by mologie** [comment]
My comment from Jun 13 in this thread shows how to change the default toolchain from command line. You can also do so from Xcode -> Preferences... -> Locations -> Command Line Tools.

**Comment by JoeyCarson** [comment]
Yes I just saw that.  I do have my xcode-select pointing to the proper directory, so if that was your solution, then mine perhaps just shows the same symptom and not the same cause.

**Comment by sinzone** [comment]
+1

**Comment by polyma** [comment]
It helps to uninstall the older version of Xcode:
    sudo <Xcode>/Library/uninstall-devtools --mode=all
    <Xcode> is the directory where the tools are installed. For typical installations the full path is      /Developer/Library/uninstall-devtools

Also freed up 4GB, wahey!

**Comment by nonocut** [comment]
+1 @mologie

**Comment by hartman** [comment]
It is with luck that I found this ticket, because the way npm install fails when this occurs is really difficult to trace back to this problem. Anyone have ideas how the error reporting of node or gyp can be improved to make this more clear ?

**Comment by cstyan** [comment]
I'm getting this error same error in OSX 10.10 using xcode 6.4, just wondering if upgrading OSX and xcode is going to be easier than tracking down the cause in OSX 10.10/xcode 6.4.

**Comment by mikemcquaid** [comment]
> I'm getting this error same error in OSX 10.10 using xcode 6.4, just wondering if upgrading OSX and xcode is going to be easier than tracking down the cause in OSX 10.10/xcode 6.4.

Yes, it will be.

**Comment by cstyan** [comment]
I simply upgraded Xcode and that fixed the issue.  So this issue is independent of OSX version.
